### PR TITLE
Revert "Support drag-and-drop for submenus of navigation blocks (#24479)"

### LINKED ISSUE #30177: Error when dragging and dropping inner blocks of the navigation block
## Description
Block drag and drop seems to be buggy in the navigation block.

When dropping a block, often the block disappears, and an error is thrown in the console:
Screenshot 2021-03-24 at 6 18 28 pm

The stack trace seems to indicate an error with one of the rich text formats.

Originally reproduce in the navigation editor, but also happens in the post editor.

## Step-by-step reproduction instructions
1. Add some inner blocks to a nav block
2. Try re-ordering them using drag and drop

## Expected behaviour
It works

## Actual behaviour
The dragged block disappears on drop

### PR DESCRIPTION
**Pull request by ellatrix**
This reverts commit 748c632feccbffb51a6e3c79972a447b71211a0c.

## Description

Fixes #30177.
Unfortunately we have to revert #24479.

After a refactoring (#29186) the custom drag behaviour of the navigation link block is throwing an infinite loop error.
If we want better drag and drop behaviour in nested contexts, we could implement this in the block editor rather than the block.

## How has this been tested?

Dragging a navigation link block within the navigation block should work again.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-list/use-block-props/use-block-class-names.js [hunk 1]
```diff
@@ -50,12 +50,12 @@ export function useBlockClassNames( clientId ) {
 				spotlightEntityBlocks
 			);
 			return classnames( 'block-editor-block-list__block', {
-				'is-selected': isSelected && ! isDragging,
```
Comment (ellatrix) [inline packages/block-editor/src/components/block-list/use-block-props/use-block-class-names.js:53]: This was a strange change. We shouldn't have to remove this class.

**Comment by jasmussen** [comment]
Thanks for the initial effort. It's important to get drag and drop right across the editor, and the more holistically we can do that, the better. That principle remains even if we need to revert this.

I'll give this more testing tomorrow, including on the navigation screen, but wanted to share a 👍.

**Review by gwwar — APPROVED** [review]
Thanks @ellatrix for debugging with me and spinning up the revert 💖 This tests well for me. I verified that we can still add items to the navigation block, add submenus, add items to the submenus, type to edit links and drag items within the navigation block without the block throwing an error or a WSOD for the editor.

For folks 👀 the PR, one thing to keep in mind for a potential second attempt is to aim for solving the general case, instead of adding custom code in the block:

> One of the most important aspects of the navigation block is that it helps define child block APIs we can use in other blocks. Any time it deviates or does something custom we lose that advantage and create an extra point of maintenance, code debt, and fragility -- @mtias 

#### Expected updated change in behavior: 

We can't drag top level items into submenu, but we can still drag submenu items into the top level.

https://user-images.githubusercontent.com/1270189/112379507-7841c200-8ca5-11eb-8c48-8fd9e8c8967e.mp4

#### Horizontal Drag

https://user-images.githubusercontent.com/1270189/112379755-c787f280-8ca5-11eb-900c-f7de0b03accb.mp4

#### Vertical Drag

https://user-images.githubusercontent.com/1270189/112379615-99a2ae00-8ca5-11eb-913f-418515936b88.mp4

https://user-images.githubusercontent.com/1270189/112379621-9d363500-8ca5-11eb-9d70-9e6d273917d5.mp4

**Comment by ellatrix** [comment]
I'm fine with fixing it with #30219 and keeping the custom behaviour. Hopefully we can make it part of the block editor some time.

**Pull request closed without merging**